### DISCUSSION TITLE
How to get the mass

### DISCUSSION THREAD

**Discussion by factal** [enable3d/enable3d#172] [Help / Questions]
Hello. I'm working on a physics engine based minigame and I need to get the mass of an object to calculate various physical quantities. However, there is no mass property in the PhysicsBody class, and PhysicsBody.ammo doesn't seem to be able to use getInvMass(), which should exist in btRigidBody. Can you please tell me how to get the mass of an object?

**Comment by yandeu** [comment]
The default mass is always 1 for dynamic objects. 

True, the mass is not stored in the physics body. You have to keep track of it somewhere else.

**Comment by luddensdesir** [comment]
What the heck? I added an object like this:

    let c2 = this.add.box({ x: 1, y: 0, z: 0, mass: 55555 });

But the mass isn't stored anywhere on the object?

I am preemptively un-upvoting my comment so you don't have to downvote.

**Reply by yandeu** [reply]
https://pybullet.org/Bullet/phpBB3/viewtopic.php?t=12802

But I don't think `getInvMass()` is available in ammo.js.

**Reply by luddensdesir** [reply]
I'm using enable3d and finally found group.body.ammo.setMassProps(mass, centerOfMass);

And it works.

**Reply by luddensdesir** [reply]
Strange. When I set mass that adding torque no longer works. :\
